thetis/tracer_eq.py

```diff
@@ -295,9 +295,11 @@ def residual(self, solution, solution_old, fields, fields_old, bnd_conditions=No
             elemsize = (self.h_elem_size*(self.normal[0]**2
                                           + self.normal[1]**2)
                         + self.v_elem_size*self.normal[2]**2)
-            sigma = 5.0*degree_h*(degree_h + 1)/elemsize
-            if degree_h == 0:
-                sigma = 1.5/elemsize
+            sigma = fields_old.get('sipg_parameter')
+            if sigma is None:
+                sigma = 5.0*degree_h*(degree_h + 1)/elemsize
```

Comment:
Note that this uses a different definition of the parameter than alpha in solver2d because here we are already dividing by a cellsize h, i.e. sigma=alpha/h (with a default of alpha=5*p*(p+1)). I think we should be setting alpha as a user parameter, as h will be spatially varying.